kopf/reactor/peering.py

```diff
@@ -79,6 +79,26 @@ def __init__(self,
     def __repr__(self):
         return f"{self.__class__.__name__}({self.id}, namespace={self.namespace}, priority={self.priority}, lastseen={self.lastseen}, lifetime={self.lifetime})"
 
+    @classmethod
+    def detect(cls,
+               standalone: bool,
+               peering: Optional[str],
+               **kwargs) -> Optional:
+        if standalone:
+            return None
+
+        if peering:
+            if Peer._is_peering_exist(peering):
+                return cls(peering=peering, **kwargs)
+            else:
+                raise Exception(f"The peering {peering} was not found")
+
+        if Peer._is_default_peering_setup():
+            return cls(peering=peering, **kwargs)
```

Comment:
Here, it causes a crash on start in auto-detection mode (when `peering is None`). Perhaps, this was intended:

```python
return cls(peering=PEERING_DEFAULT_NAME, **kwargs)
```

Otherwise, it passes `None`, and then it remembers `ourselves.peering=None`, and cannot perform `PATCH` on `None`, and fails on start in the auto-detection mode.

Reply:
Ah true, I didn't test this path manually 👍 
merged.